app/pages/lab/actions/workflow.js

```diff
@@ -3,12 +3,23 @@ var apiClient = require('panoptes-client/lib/api-client');
 var workflows = apiClient.type('workflows');
 
 var workflowActions = {
-  createWorkflowForProject: function(projectID, workflowData) {
+  createWorkflowForProject: function(project, workflowData) {
     var allWorkflowData = Object.assign({
-      links: {project: projectID}
+      links: {project: project.id}
     }, workflowData);
 
     return workflows.create(allWorkflowData).save();
+  },
+  copyWorkflowForProject: async function(project, workflowData) {
+    await project.addLink('workflows', [workflowData.id]);
+    project.uncacheLink('workflows');
+    // TODO: does this sort work for this query?
+    const newWorkflows = await project.get('workflows', { sort: '-updated_at' });
+    // TODO: is there a better way to get the ID of the new workflow?
+    const newWorkflowID = Math.max(...newWorkflows.map(workflow => workflow.id));
```

Comment:
This is terrible and we can make this explicit in the create links response, see above.

Reply:
If `project.links.workflows` was in a predictable order, I could pluck the last workflow from the array. 

I could also compare the new links array with the old one and take the difference.

Reply:
This code is gone now. 🙏